app/config-setup.py

```diff
@@ -0,0 +1,28 @@
+import json
+import requests
+
+open('/home/pi/config.json', 'w').write('{}')
+config = json.load(open('/home/pi/config.json', 'r'))
+
+def get_box_id():
+    response = requests.get('http://trinity-stroud.hedgehogrepublic.org/api/locations/').json()
+    location_list = response['results']
+    i = 0
+    for j in location_list:
+        i += 1
+        print(str(i) + ') ' + j["name"])
+    box = int(input('Box Number: '))
+    config['box_id'] = location_list[box - 1]['code']
+    
+
+
+def main():
+    print('Welcome to the HogPi Setup')
+    try:
+        get_box_id()
+    except Exception:
```

Comment:
In general, you should never catch with a "bare except". Two SO answers here explain why [1](https://stackoverflow.com/a/14797508), [2](https://stackoverflow.com/a/14797463).

I can't suggest a better exception because I don't understand why you'd have a try/except here anyway. Can you explain?

Reply:
If someone inputs an invalid box number, it will raise an IndexError, the try except block is there to catch this if it happens and if it does re-runs the function to try again